### PR TITLE
Moves to nebula.netflixoss 2.2.5 and fixes broken build

### PR DESCRIPTION
**Pull request by adriancole**

### CODE REVIEW

**Comment by adriancole** [comment]
@rspieldenner PTAL

File: cli/src/main/java/denominator/cli/Denominator.java [hunk 1]
```diff
@@ -136,7 +138,7 @@ public static String providerAndCredentialsTable() {
             StringBuilder builder = new StringBuilder();
 
             builder.append(format(table, "provider", "url", "duplicateZones", "credentialType", "credentialArgs"));
-            for (Provider provider : Providers.list()) {
+            for (Provider provider : ImmutableSortedSet.copyOf(Ordering.usingToString(), Providers.list())) {
```
Comment (adriancole) [inline cli/src/main/java/denominator/cli/Denominator.java]: led to flakey tests

File: cli/src/main/java/denominator/cli/Denominator.java [hunk 2]
```diff
@@ -265,13 +267,14 @@ String getFileContentsFromPath(String path) throws IOException {
...
 
         Map<String, ?> getConfigFromEnv() {
-            Map<String, Object> env = new HashMap<String, Object>();
-            String provider = getEnvValue("PROVIDER");
-            env.put("provider", provider);
-            env.put("url", getEnvValue("URL"));
-
-            Map<String, String> credentialMap = new HashMap<String, String>();
-            Provider providerLoaded = Providers.getByName(provider);
+            Map<String, Object> env = new LinkedHashMap<String, Object>();
```
Comment (adriancole) [inline cli/src/main/java/denominator/cli/Denominator.java]: this code block was a mess and also broke tests

File: cli/src/test/java/denominator/cli/DenominatorTest.java [hunk 3]
```diff
@@ -83,13 +89,13 @@ public void testCredentialsFromYaml() {
...
 
     private String getTestYaml() {
-        return "providerConfigurationName: blah1\n" +
+        return "name: blah1\n" +
             "provider: route53\n" +
             "credentials:\n" +
             "  accessKey: foo1\n" +
             "  secretKey: foo2\n" +
             "---\n" +
-            "providerConfigurationName: blah2\n" +
```
Comment (adriancole) [inline cli/src/test/java/denominator/cli/DenominatorTest.java]: someone's find/replace broke a bunch of tests

File: discoverydns/src/test/java/denominator/discoverydns/crypto/PemsTest.java [hunk 4]
```diff
@@ -35,15 +32,6 @@ public void readPrivateKeyRsaUnencrypted() throws IOException {
         assertNotNull(key);
         assertEquals(key.getAlgorithm(), "RSA");
         assertEquals(key.getFormat(), "PKCS#8");
-        assertTrue(key.toString().contains("13027056791347954860993705134784360"
```
Comment (adriancole) [inline discoverydns/src/test/java/denominator/discoverydns/crypto/PemsTest.java]: toString isn't consistent across JVMs. Not worth testing for.

**Comment by adriancole** [comment]
@rspieldenner sorry to say I might need your help to update the pull request builder to a new JDK. Looks like the failure was language level-related.

**Comment by rspieldenner** [comment]
Changed job to jdk7. Still test failures.

**Comment by adriancole** [comment]
Mind cleaning the workspace? The assertions that failed in that job are at
the wrong line numbers.. maybe jenkins is behind this PR?

Locally JDK 7 and 8 pass. The failures above are the reason for the "fix
build" commit (i.e. denominator master has been broken a while until this)

**Comment by rspieldenner** [comment]
I don't know why the relaunch from cloudbees isn't rebuilding this branch. Do you mind just amending the last commit and pushing it again?

**Comment by adriancole** [comment]
@rspieldenner dunno why, but the PR builder seems stuck at commit 3c0da6f6a7f88df059df320563523f5a4a3372aa which is broke.. maybe I should just merge since it passes locally

**Comment by adriancole** [comment]
@rspieldenner wires crossed.. just force pushed after rewording a commit.

**Comment by adriancole** [comment]
whoot!

**Comment by adriancole** [comment]
cherry-picked.. 4.x is now back in business

**Pull request merged**